### ISSUE TITLE
Error: package or namespace load failed for ‘garchmodels’

### ISSUE THREAD

**Issue by dgslv** [AlbertoAlmuinha/garchmodels#8]
Already tried to reinstall R, Rstudio and all packages but still getting this error...

```
library(garchmodels)
Loading required package: parsnip
Loading required package: rugarch
Loading required package: parallel

Attaching package: ‘rugarch’

The following object is masked from ‘package:stats’:

    sigma

Loading required package: rmgarch
Error: package or namespace load failed for ‘garchmodels’:
 .onLoad failed in loadNamespace() for 'garchmodels', details:
  call: check_model_doesnt_exist(model)
  error: Model `garch_reg` already exists
```

**Comment by dgslv** [comment]
I'm running on macOS Big Sur (11.6)

**Comment by dgslv** [comment]
if I try again then the error changes to the following:

```
Error: package or namespace load failed for ‘garchmodels’:
 .onLoad failed in loadNamespace() for 'garchmodels', details:
  call: check_model_doesnt_exist(model)
  error: Model `garch_reg` already exists
```

**Comment by zhushazang** [comment]
Hello there

Using R:
platform       x86_64-pc-linux-gnu         
arch           x86_64                      
os             linux-gnu                   
system         x86_64, linux-gnu           
status                                     
major          4                           
minor          1.3                         
year           2022                        
month          03                          
day            10                          
svn rev        81868                       
language       R                           
version.string R version 4.1.3 (2022-03-10)
nickname       One Push-Up                

Inside Linux: Linux asgard 5.16.14-arch1-1 #1 SMP PREEMPT Fri, 11 Mar 2022 17:40:36 +0000 x86_64 GNU/Linux

> sessionInfo()
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Arch Linux

Matrix products: default
BLAS/LAPACK: /usr/lib/libopenblas_piledriverp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.utf8       LC_NUMERIC=C             
 [3] LC_TIME=en_US.utf8        LC_COLLATE=C             
 [5] LC_MONETARY=en_US.utf8    LC_MESSAGES=en_US.utf8   
 [7] LC_PAPER=en_US.utf8       LC_NAME=C                
 [9] LC_ADDRESS=C              LC_TELEPHONE=C           
[11] LC_MEASUREMENT=en_US.utf8 LC_IDENTIFICATION=C      

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.1.3 tools_4.1.3   


I was able to reproduce the error above when trying to install the package:

> install.packages('garchmodels',repos="http://cran.r-project.org",checkBuik=TRUE,Ncpus=8)
Installing package into ‘/home/rodolfo/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
trying URL 'http://cran.r-project.org/src/contrib/garchmodels_0.1.1.tar.gz'
Content type 'application/x-gzip' length 2821550 bytes (2.7 MB)
==================================================
downloaded 2.7 MB

* installing *source* package ‘garchmodels’ ...
** package ‘garchmodels’ successfully unpacked and MD5 sums checked
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘garchmodels’:
 .onLoad failed in loadNamespace() for 'garchmodels', details:
  call: check_mode_for_new_engine(model, eng, mode)
  error: 'regression' is not a known mode for model `garch_reg()`.
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/rodolfo/R/x86_64-pc-linux-gnu-library/4.1/garchmodels’

The downloaded source packages are in
        ‘/tmp/Rtmp3xIsXI/downloaded_packages’
Warning message:
In install.packages("garchmodels", repos = "http://cran.r-project.org",  :
  installation of package ‘garchmodels’ had non-zero exit status

**Comment by dgslv** [comment]
here follows my `sessionInfo()`

```
> sessionInfo()
R version 4.1.3 (2022-03-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.6

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
  [1] spd_2.0-1                   xts_0.12.1                  lubridate_1.8.0             DiceDesign_1.9             
  [5] numDeriv_2016.8-1.1         tools_4.1.3                 utf8_1.2.2                  R6_2.5.1                   
  [9] rugarch_1.4-7               rpart_4.1.16                KernSmooth_2.23-20          DBI_1.1.2                  
 [13] colorspace_2.0-3            yardstick_0.0.9             nnet_7.3-17                 withr_2.5.0                
 [17] tidyselect_1.1.2            bit_4.0.4                   compiler_4.1.3              cli_3.2.0                  
 [21] scales_1.1.1                mvtnorm_1.1-3               tune_0.1.6                  digest_0.6.29              
 [25] rmarkdown_2.13              pkgconfig_2.0.3             htmltools_0.5.2             parallelly_1.30.0          
 [29] lhs_1.1.4                   fastmap_1.1.0               rlang_1.0.2                 generics_0.1.2             
 [33] zoo_1.8-9                   mclust_5.4.9                dplyr_1.0.8                 magrittr_2.0.2             
 [37] Matrix_1.4-0                Rcpp_1.0.8.2                munsell_0.5.0               fansi_1.0.2                
 [41] GPfit_1.0-8                 lifecycle_1.0.1             furrr_0.2.3                 yaml_2.3.5                 
 [45] pROC_1.18.0                 MASS_7.3-55                 plyr_1.8.6                  recipes_0.2.0              
 [49] grid_4.1.3                  parallel_4.1.3              listenv_0.8.0               rmgarch_1.3-9              
 [53] crayon_1.5.0                lattice_0.20-45             SkewHyperbolic_0.4-0        splines_4.1.3              
 [57] knitr_1.37                  pillar_1.7.0                dials_0.1.0                 corpcor_1.6.10             
 [61] future.apply_1.8.1          codetools_0.2-18            parsnip_0.2.0               timetk_2.7.0               
 [65] glue_1.6.2                  evaluate_0.15               rsample_0.1.1               DistributionUtils_0.6-0    
 [69] vctrs_0.3.8                 foreach_1.5.2               gtable_0.3.0                purrr_0.3.4                
 [73] tidyr_1.2.0                 future_1.24.0               assertthat_0.2.1            ks_1.13.4                  
 [77] ggplot2_3.3.5               xfun_0.30                   gower_1.0.0                 prodlim_2019.11.13         
 [81] Rmpfr_0.8-7                 pracma_2.3.8                ff_4.0.5                    GeneralizedHyperbolic_0.8-4
 [85] Rsolnp_1.16                 class_7.3-20                survival_3.2-13             pcaPP_1.9-74               
 [89] timeDate_3043.102           truncnorm_1.0-8             tibble_3.1.6                iterators_1.0.14           
 [93] hardhat_0.2.0               lava_1.6.10                 Bessel_0.6-0                workflows_0.2.5            
 [97] gmp_0.6-4                   globals_0.14.0              ellipsis_0.3.2              ipred_0.9-12    
 ```

**Comment by sreedta8** [comment]
Unable to install ```garchmodels``` using ```devtools``` from the GitHub repository.

OS: Lubuntu (Impish Idri 21.10)

**Installation and Error Output**

```
> devtools::install_github("AlbertoAlmuinha/garchmodels")
Downloading GitHub repo AlbertoAlmuinha/garchmodels@HEAD
✔  checking for file ‘/tmp/RtmpxN9XTa/remotes1edd3fb6d6e8/AlbertoAlmuinha-garchmodels-2385a06/DESCRIPTION’ ...
─  preparing ‘garchmodels’:
✔  checking DESCRIPTION meta-information ...
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘garchmodels_0.1.1.tar.gz’
   
Installing package into ‘/home/sreedta/R/x86_64-pc-linux-gnu-library/4.1’
(as ‘lib’ is unspecified)
* installing *source* package ‘garchmodels’ ...
** using staged installation
** R
** data
*** moving datasets to lazyload DB
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
*** copying figures
** building package indices
** installing vignettes
** testing if installed package can be loaded from temporary location
Error: package or namespace load failed for ‘garchmodels’:
 .onLoad failed in loadNamespace() for 'garchmodels', details:
  call: check_mode_for_new_engine(model, eng, mode)
  error: 'regression' is not a known mode for model `garch_reg()`.
Error: loading failed
Execution halted
ERROR: loading failed
* removing ‘/home/sreedta/R/x86_64-pc-linux-gnu-library/4.1/garchmodels’
* restoring previous ‘/home/sreedta/R/x86_64-pc-linux-gnu-library/4.1/garchmodels’
Warning message:
In i.p(...) :
  installation of package ‘/tmp/RtmpxN9XTa/file1edd7a4a3e9/garchmodels_0.1.1.tar.gz’ had non-zero exit status
```
**Session Info**
```
> sessionInfo()
R version 4.1.3 (2022-03-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 21.10

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ctv_0.9-2

loaded via a namespace (and not attached):
 [1] magrittr_2.0.2    usethis_2.1.5     devtools_2.4.3    pkgload_1.2.4    
 [5] R6_2.5.1          rlang_1.0.2       fastmap_1.1.0     tools_4.1.3      
 [9] pkgbuild_1.3.1    sessioninfo_1.2.2 cli_3.2.0         withr_2.5.0      
[13] ellipsis_0.3.2    remotes_2.4.2     rprojroot_2.0.2   lifecycle_1.0.1  
[17] crayon_1.5.0      brio_1.1.3        processx_3.5.2    purrr_0.3.4      
[21] callr_3.7.0       fs_1.5.2          ps_1.6.0          curl_4.3.2       
[25] testthat_3.1.2    memoise_2.0.1     glue_1.6.2        cachem_1.0.6     
[29] compiler_4.1.3    desc_1.4.0        prettyunits_1.1.1
```

**Comment by dgslv** [comment]
should be fixed by #7.
